setup.py

```diff
@@ -0,0 +1,63 @@
+    entry_points={"console_scripts": ["wow-addon-updater = updater.__main__:main"],},
+    extras_require={
+        "dev": [
+            "appdirs==1.4.4",
+            "attrs==19.3.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "black==19.10b0; python_version >= '3.6'",
+            "cached-property==1.5.1",
+            "cerberus==1.3.2",
+            "certifi==2020.6.20",
+            "chardet==3.0.4",
+            "click==7.1.2; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'",
+            "colorama==0.4.3; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'",
+            "coverage==5.2.1",
+            "distlib==0.3.1",
+            "idna==2.10; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "orderedmultidict==1.0.1",
+            "packaging==20.4; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "pathspec==0.8.0",
+            "pep517==0.8.2",
+            "pip-shims==0.5.3; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'",
+            "pipenv-setup==3.1.1",
+            "pipfile==0.0.2",
+            "plette[validation]==0.2.3; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "pyparsing==2.4.7; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "python-dateutil==2.8.1; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "regex==2020.7.14",
+            "requests==2.24.0",
+            "requirementslib==1.5.12; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'",
+            "six==1.15.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "toml==0.10.1",
+            "tomlkit==0.7.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'",
+            "typed-ast==1.4.1",
+            "urllib3==1.25.10; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4' and python_version < '4'",
+            "vistir==0.5.2; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+            "wheel==0.34.2; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'",
+        ]
+    },
+    install_requires=[
+        "beautifulsoup4==4.9.1",
+        "certifi==2020.6.20",
+        "chardet==3.0.4",
+        "cloudscraper==1.2.46",
+        "idna==2.10; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+        "pyparsing==2.4.7; python_version >= '2.6' and python_version not in '3.0, 3.1, 3.2, 3.3'",
+        "requests==2.24.0",
+        "requests-toolbelt==0.9.1",
+        "soupsieve==2.0.1; python_version >= '3.5'",
+        "urllib3==1.25.10; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4' and python_version < '4'",
+    ],
```

Comment:
A little concerned about what this actually generated. On some platforms, I've run into issues with the SSL version used when communicating with C*oudflare (trying to remember, I think the culprit was Windows), so there's a soft requirement on Python 3.8 there. Do the lines like `python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4' and python_version < '4'"` enable too loose of a version requirement? Is there a way to require a Python version for the package consumer?

Reply:
Maybe with `python_requires='>=3.8.0'` in `setup.py`?

Reply:
Exactly your second comment. [python_requires](https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires) allows for exactly that, or you can technically do the same thing (with different strings) as above. Just `python_requires='>=3.8.0'`.

3.6.x is still fairly common, but as you're saying if it really is a hard requirement due to other deps 3.8 is fine and readily available.

I can toss that in for sure.